packages/cli/src/dbt/targets/Databricks/oauth.ts

```diff
@@ -0,0 +1,269 @@
+import { generators } from 'openid-client';
+import { URL } from 'url';
+import { promisify } from 'util';
+import GlobalState from '../../../globalState';
+
+const execAsync = promisify(exec);
+
+// Helper function to open browser
+const openBrowser = async (url: string): Promise<void> => {
+    try {
+        const { platform } = process;
+
+        if (platform === 'darwin') {
+            await execAsync(`open "${url}"`);
+        } else if (platform === 'win32') {
+            await execAsync(`start "${url}"`);
+        } else {
+            await execAsync(`xdg-open "${url}"`);
+        }
+    } catch (error) {
+        GlobalState.debug(`> Could not open browser automatically: ${error}`);
+    }
+};
+
+/**
+ * Databricks OAuth tokens result
+ */
+export interface DatabricksOAuthTokens {
+    accessToken: string;
+    refreshToken: string;
+    expiresAt: number; // Unix timestamp in seconds
+}
+
+/**
+ * Perform Databricks U2M OAuth flow
+ * Opens browser for user authentication and exchanges authorization code for tokens
+ * @param host Databricks workspace host
+ * @param clientId OAuth client ID (defaults to 'databricks-cli')
+ */
+export const performDatabricksOAuthFlow = async (
+    host: string,
+    clientId: string,
+    clientSecret: string | undefined,
+): Promise<DatabricksOAuthTokens> => {
+    // Create a promise that will be resolved when we get the authorization code
+    let resolveAuth: (value: { code: string; state: string }) => void;
+    let rejectAuth: (reason: Error) => void;
+    const authPromise = new Promise<{ code: string; state: string }>(
+        (resolve, reject) => {
+            resolveAuth = resolve;
+            rejectAuth = reject;
+        },
+    );
+    let port = 0;
+
+    // Generate PKCE values
+    const codeVerifier = generators.codeVerifier();
+    const codeChallenge = generators.codeChallenge(codeVerifier);
+    const state = generators.state();
+
+    // Create HTTP server to handle the callback at root path
+    const server = http.createServer((req, res) => {
+        const callbackUrl = new URL(req.url || '/', `http://localhost:${port}`);
+        const code = callbackUrl.searchParams.get('code');
+        const returnedState = callbackUrl.searchParams.get('state');
+        const error = callbackUrl.searchParams.get('error');
+
+        res.setHeader('Content-Type', 'text/html');
+
+        if (error === 'access_denied') {
+            rejectAuth(new AuthorizationError(`OAuth error: access denied`));
+            res.writeHead(400);
+            res.end(
+                '<html><body><h1>Authentication Failed</h1><p>Access denied. You can close this window.</p></body></html>',
+            );
+            return;
+        }
+        if (error) {
+            rejectAuth(new AuthorizationError(`OAuth error: ${error}`));
+            res.writeHead(400);
+            res.end(
+                `<html><body><h1>Authentication Failed</h1><p>Error: ${error}</p></body></html>`,
```

Comment:
Is this all a common pattern we can abstract for all our browser based Oauth flows in the CLI?